stepper.c

```diff
@@ -519,36 +519,149 @@ void st_reset()
   ESTOP_DDR  &= ~(ESTOP_BIT);           //estop as input
   ESTOP_PORT |= (1<<RUN_ENABLE_BIT);  //allow motors to run
   ESTOP_PORT &= ~(ESTOP_BIT);           //estop input normal-low
 
-  //Microstepping
+  // Microstepping
   MS_DDR = MS_MASK; //all output
 
   MS_PORT = settings.microsteps&MS_MASK;
-  //Phase Current Decay. 
+
+  // Phase Current Decay
   PFD_DDR = PFD_MASK; //all output
-  uint8_t axis,value=0;
-  for (axis=0;axis<N_AXIS;axis++) {
-    value|=SET_DECAY_MODE(axis,settings.decay_mode);
+  uint8_t axis, value=0;
+  for (axis=0; axis<N_AXIS; axis++) {
+    value |= SET_DECAY_MODE(axis, settings.decay_mode);
   }
-  PFD_PORT = value&PFD_MASK;
+  PFD_PORT = value & PFD_MASK;
 
-  //PORTA  for voltage sensors
-  //TODO -sset pins 0..3 as input.  Read on command.  (create command)
+  // MVOLT is used for voltage sensing
+  // TODO:  read on command.  (create command)
   MVOLT_DDR &= ~(MVOLT_MASK);
-  MVOLT_PORT |= MVOLT_MASK; //internal pull-up, normal high  //TODO: should have been analog?
+  MVOLT_PORT |= MVOLT_MASK; // internal pull-up, normal high
+
+  // FDBK is for additional 1 voltage sensor (gripper force feedback)
+  // Feedback sensor voltage is called FVOLT
+  FVOLT_DDR |= ~(FVOLT_MASK);
+  FVOLT_PORT |= FVOLT_MASK;  // internal pull-up????  it's an analog
+
+  // Setup CCTRL (current control) ports
+  // TODO: make this output PWMs with configurable values
+  // for now, just turning on to full power
+  CCTRL_DDR |= (1<<CCTRL_CG_BIT);
+  CCTRL_DDR |= (1<<CCTRL_XY_BIT);
+
+  // TODO: Replace this with analog out code
+  CCTRL_PORT |= (1<<CCTRL_CG_BIT);
+  CCTRL_PORT |= (1<<CCTRL_XY_BIT);
 
-  //Setup IO Reset Port
+  // Setup IO Reset Port
   IO_RESET_DDR |= IO_RESET_MASK;
   IO_RESET_PORT &= ~IO_RESET_MASK; //don't reset
+
+
+  // Setup hardware PWM timing on Timer3 to control OC3 A,B and C
+  // with pwm signals
+
+  // PWM Setup, fast pwm mode, 8 bit:
+  //  WGM33=0, on register B
+  //  WGM32=1, on register B
+  //  WGM31=0, on register A
+  //  WGM30=1, on register A
+
+  // No prescaling to get fastest speed
+  //  CS32=0, register B
+  //  CS31=0, register B
+  //  CS30=1, register B
+
+  // Enable non inverting fast PWM compare
+  //  Enable all three comparators (A, B, C)
+  //  COM3A1=1, on register A
+  //  COM3A0=0, on register A
+  //  COM3B1=1, on register A
+  //  COM3B0=0, on register A
+  //  COM3C1=1, on register A
+  //  COM3C0=0, on register A
+
+  TCCR3A = (1<<COM3A1)+(0<<COM3A0) + (1<<COM3B1)+(0<<COM3B0) + (1<<COM3C1)+(0<<COM3C0) + (0<<WGM31)+(1<<WGM30);
+  TCCR3B = (0<<WGM33)+(1<<WGM32) + (0<<CS32)+(0<<CS31)+(1<<CS30);
+  TCCR3C = 0;
+
+  // Set all three pins as OUTPUT
+  PWM_OUT_DDR |= 1<<PWM_OUT_FSENSE_BIT;
+  PWM_OUT_DDR |= 1<<PWM_OUT_XY_CTRL_BIT;
+  PWM_OUT_DDR |= 1<<PWM_OUT_CG_CTRL_BIT;
+
+  // Set OLD PWM Sensor Drive as INPUT so it's not driving anything
+  // TODO: remove this as un-needed on Rev5 boards
+  FSCTRL_DDR |= 0<<FSCTRL_BIT;
```

Comment:
In board conditional is hard.  I know the IO board can see that voltage divider.  I don't know if we have it connected to an AIN on motion.